akka-projection-slick/src/test/scala/akka/projection/slick/SlickProjectionSpec.scala

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.projection.slick
+
+import java.util.UUID
+
+import akka.stream.scaladsl.Source
+import akka.{ Done, NotUsed }
+import com.typesafe.config.{ Config, ConfigFactory }
+import org.scalatest.OptionValues
+import org.slf4j.LoggerFactory
+import slick.basic.DatabaseConfig
+import slick.dbio.DBIOAction
+import slick.jdbc.H2Profile
+
+import scala.concurrent.Future
+import scala.concurrent.duration._
+
+object SlickProjectionSpec {
+  def config: Config = ConfigFactory.parseString("""
+      |akka.projection.slick = {
+      |
+      |  profile = "slick.jdbc.H2Profile$"
+      |
+      |  # TODO: configure connection pool and slick async executor
+      |  db = {
+      |   url = "jdbc:h2:mem:test1"
+      |   driver = org.h2.Driver
+      |   connectionPool = disabled
+      |   keepAliveConnection = true
+      |  }
+      |}
+      |""".stripMargin)
+}
+
+class SlickProjectionSpec extends SlickSpec(SlickProjectionSpec.config) with OptionValues {
+
+  val logger = LoggerFactory.getLogger(this.getClass)
+
+  implicit val patience: PatienceConfig = PatienceConfig(10.seconds, 500.millis)
+
+  val repository = new TestRepository(dbConfig)
+
+  override protected def beforeAll(): Unit = {
+    super.beforeAll()
+    repository.createTable()
+  }
+
+  "A Slick projection" must {
+
+    "persist events and offset in same transactional" in {
+
+      implicit val dispatcher = actorSystem.dispatcher
+
+      val projectionId = UUID.randomUUID().toString
+      val entityId = UUID.randomUUID().toString
+
+      withClue("check - offset is empty") {
+        val offsetOpt = offsetStore.readOffset[Long](projectionId).futureValue
+        offsetOpt shouldBe empty
+      }
+
+      val slickProjection =
+        SlickProjection.transactional(
+          projectionId = projectionId,
+          sourceProvider = sourceProvider(entityId),
+          offsetExtractor = offsetExtractor,
+          databaseConfig = dbConfig) { envelope => repository.concatToText(envelope.id, envelope.message) }
+
+      runProjection(slickProjection) {
+        eventually {
+          withClue("check - all values were concatenated") {
+            val concatStr = dbConfig.db.run(repository.findById(entityId)).futureValue.value
+            concatStr.text shouldBe "abc|def|ghi|jkl|mno|pqr"
+          }
```

Comment:
This will be a common scenario when for the Projection TestKit. You setup a projection, you run it and check that the results will eventually be reflected on the storage. 

There is room for improvement here. Instead of wrapping it on a `Scalatest.eventually` we should have the re-try built-in, like Akka's `within` (with or without dilated time). 

We should also take into considerations that users may build a projection for an infinite Source, for example, Akka Query eventsByTag. In such a case, if the passed assertion blocks succeeds or if it fail after some timeout we can stop the projection. 

Anyway, for later.